src/toolbars/SelectionBar.cpp

```diff
@@ -167,42 +175,103 @@ void SelectionBar::Create(wxWindow * parent)
    UpdatePrefs();
 }
 
+AButton* SelectionBar::MakeSetupButton()
+{
+   wxImage up = theTheme.Image(bmpRecoloredUpSmall);
+   up.Rescale(20, 20, wxIMAGE_QUALITY_HIGH);
+   wxImage down = theTheme.Image(bmpRecoloredDownSmall);
+   down.Rescale(20, 20, wxIMAGE_QUALITY_HIGH);
+   wxImage hiliteUp = theTheme.Image(bmpRecoloredUpHiliteSmall);
+   hiliteUp.Rescale(20, 20, wxIMAGE_QUALITY_HIGH);
+   wxImage hiliteDown = theTheme.Image(bmpRecoloredHiliteSmall);
+   hiliteDown.Rescale(20, 20, wxIMAGE_QUALITY_HIGH);
+   
+   auto btn = safenew AButton(
+      this, wxID_ANY, wxDefaultPosition, wxSize { 20, 20 }, up, hiliteUp, down,
+      hiliteDown, up, false);
+   
+   btn->SetButtonType(AButton::FrameButton);
+   btn->SetIcon(theTheme.Image(bmpCogwheel));
+   btn->SetLabel({});
+   btn->SetName(XO("Selection Toolbar Setup").Translation());
+
+   return btn;
+}
 
-auStaticText * SelectionBar::AddTitle(
+void SelectionBar::AddTitle(
    const TranslatableString & Title, wxSizer * pSizer ){
    const auto translated = Title.Translation();
    auStaticText * pTitle = safenew auStaticText(this, translated );
    pTitle->SetBackgroundColour( theTheme.Colour( clrMedium ));
    pTitle->SetForegroundColour( theTheme.Colour( clrTrackPanelText ) );
    pSizer->Add( pTitle, 0, wxEXPAND | wxRIGHT, 5 );
-
-   return pTitle;
 }
 
 
-NumericTextCtrl * SelectionBar::AddTime(
-   const TranslatableString &Name, int id, wxSizer * pSizer ){
+void SelectionBar::AddTime(
+   int id, wxSizer * pSizer ){
    auto formatName = mListener ? mListener->AS_GetSelectionFormat()
       : NumericFormatSymbol{};
    auto pCtrl = safenew NumericTextCtrl(
       this, id, NumericConverter::TIME, formatName, 0.0, mRate);
-   pCtrl->SetName( Name );
+
+   pCtrl->Bind(
+      wxEVT_TEXT,
+      [this, id](auto& evt) { ModifySelection(id, evt.GetInt() != 0); });
+   
    pSizer->Add(pCtrl, 0, wxALIGN_TOP | wxRIGHT, 5);
-   return pCtrl;
+
+   mTimeControls[id] = pCtrl;
 }
 
-void SelectionBar::AddVLine(  wxSizer * pSizer ){
-   pSizer->Add(safenew wxStaticLine(this, -1, wxDefaultPosition,
-                                   wxSize(1, toolbarSingle-10),
-                                   wxLI_VERTICAL),
-                  0,  wxALIGN_CENTER_VERTICAL | wxRIGHT, 5);
+void SelectionBar::AddSelectionSetupButton(wxSizer* pSizer)
+{
+   auto setupBtn = MakeSetupButton();
+
+   auto showMenu = [this, setupBtn]()
```

Comment:
The commit that does all of this -- should be squashed with the commit two previous, and the commit that adds images should be before that, If you want to eliminate the unfinished intermediate step of the history.

Reply:
I've tried to keep history consistent and easy for review. I failed horribly in the end anyway, because nice history never matches the actual history.

Reply:
It's not "horrible"